src/UserSpaceInstrumentation/AllocateInTracee.cpp

```diff
@@ -100,34 +106,50 @@ namespace {
       nullptr, [&original_registers](void* /*ptr*/) {
         auto restore_registers_result = original_registers.RestoreRegisters();
         if (restore_registers_result.has_error()) {
-          ORBIT_FATAL("Unable to restore register state of tracee: %s",
+          ORBIT_ERROR("Unable to restore register state of tracee: %s",
                       restore_registers_result.error().message());
         }
       }};
 
+  // The system call could cause the thread to be killed, so we need to read the seccomp mode
+  // before actually executing the system call.
+  const std::optional<int> seccomp_mode = ReadSeccompModeOfThread(pid);
```

Comment:
Yes, this was planned for the next change, but we can do it in this one. The reason for doing it in another change was that I wanted to think about where to place the code that handles `SECCOMP_MODE_STRICT`. We can have it here, but if we see that the target is in that seccomp mode, we might not want to even start the injection process (in other words, not even call `ptrace`). So this could be all the way up in `InstrumentationManager::InstrumentProcess`. On the other hand, we also don't check for a debugger right at the beginning, so having this verification localized in `SyscallInTracee`, where the problem would come up, can also make sense. So yeah, this is what I wanted to discuss first. Thoughts?

Reply:
Okay, great, let's postpone this to another change. We might want to iterate through the threads and find one that is not locked down (we talked about this before ... not sure how useful this is).